source/ocs/ocs.ts

```diff
@@ -53,17 +134,717 @@ export async function getUser(
+          break;
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
     }
 
-    user = null;
+    success = rejectWithOcsError(`Unable to ${isEnabled ? 'enable' : 'disable'} user`, reason, userId, statusCode);
   }
 
-  return user;
+  return success;
+}
+
+export async function editUser(
+  connection: OcsConnection,
+  userId: string,
+  field: OcsEditUserField,
+  value: string
+): Promise<boolean> {
+  let userEdited: Promise<boolean>;
+
+  try {
+    userEdited = await promisifiedEditUser.call(connection, userId, field, value);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+
+    if (
+      (error.code === 400 || error.code === 401) &&
+      error.meta && error.meta.statuscode
+    ) {
+      switch (error.meta.statuscode) {
+        case 101:
+          reason = 'user not found';
+          break;
+        case 997:
+          reason = 'possible reasons: Does it exist? Do you have the right permissions? Is the field valid?';
+          break;
+        case 102:
+        case 103:
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
+    }
+
+    userEdited = rejectWithOcsError('Unable to edit user', reason, userId, statusCode);
+  }
+
+  return userEdited;
+}
+
+export async function getUserGroups(
+  connection: OcsConnection,
+  userId: string
+): Promise<string[]> {
+  let groups: Promise<string[]>;
+
+  try {
+    groups = await promisifiedGetUserGroups.call(connection, userId);
+  } catch (error) {
+    groups = rejectWithOcsError('Unable to get groups for user', error.message, userId);
+  }
+
+  return groups;
+}
+
+export async function getUserSubAdmins(
+  connection: OcsConnection,
+  userId: string
+): Promise<string[]> {
+  let subAdmins: Promise<string[]>;
+
+  try {
+    subAdmins = await promisifiedGetUserSubAdmins.call(connection, userId);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+
+    if (error.code === 400 && error.meta && error.meta.statuscode) {
+      switch (error.meta.statuscode) {
+        case 101:
+          reason = 'user does not exist';
+          break;
+        case 102:
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
+    }
+
+    subAdmins = rejectWithOcsError('Unable to get sub-admins for user', reason, userId, statusCode);
+  }
+
+  return subAdmins;
+}
+
+export async function resendUserWelcomeEmail(
+  connection: OcsConnection,
+  userId: string
+): Promise<boolean> {
+    let success: Promise<boolean>;
+
+    try {
+      success = await promisifiedResendUserWelcomeEmail.call(connection, userId);
+    } catch (error) {
+      let reason = error.message;
+      let statusCode = '';
+
+      if (error.code === 400 && error.meta && error.meta.statuscode) {
```

Comment:
Still… some way of creating a function that encapsulates the try/catch and the switch (as a key lookup) to which you'd pass the function that calls the API and an object of status codes and their respective reasons as well as the generic rejection message?

Reply:
I've updated the logic so it is a lot cleaner. 
I'm unable to make a generic function (like the one used in the WebDAV calls) as there is more information in the error messages based on the parameters (and I don't know how to do it).
I hope I've fixed it to your satisfaction :)